File: MsgCommunicator/DatabasePack/Demos/MySQL/C++ Builder 2009/Client/uRegister.cpp
//---------------------------------------------------------------------------
#include <vcl.h>
#pragma hdrstop

#include "uRegister.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TfmRegister *fmRegister;
//---------------------------------------------------------------------------
__fastcall TfmRegister::TfmRegister(TComponent* Owner)
  : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TfmRegister::FormClose(TObject *Sender,
      TCloseAction &Action)
{
  if (!FClose)
   Action = Forms::caNone;
}
//---------------------------------------------------------------------------
void __fastcall TfmRegister::FormShow(TObject *Sender)
{
  edHost->Text = fmMain->MsgClient1->ConnectionParams->RemoteHost;
  edPort->Text = IntToStr((int)fmMain->MsgClient1->ConnectionParams->RemotePort);
  FClose = false;
  ModalResult = mrCancel;  
}
//---------------------------------------------------------------------------
void __fastcall TfmRegister::bnCancelClick(TObject *Sender)
{
  ModalResult = mrCancel;
  FClose = true;  
}
//---------------------------------------------------------------------------
void __fastcall TfmRegister::bnRegisterClick(TObject *Sender)
{
  TMsgUserInfo UserInfo;
  int res;
  // clear UserInfo
  memset(&UserInfo,0x00,sizeof(UserInfo));
  ModalResult = mrOk;
  fmMain->MsgClient1->ConnectionParams->RemoteHost = edHost->Text;
  // if UserID empty  UserID will be assigned by the server
  if (RegUserID->Text == "")
   UserInfo.UserID = MSG_INVALID_USER_ID;
  else
   try
   {
    UserInfo.UserID = StrToIntDef(RegUserID->Text,MSG_INVALID_USER_ID);
   }
   catch (...)
   {
    ShowMessage("Invalid UserID");
    return;
   }
  try
  {
    fmMain->MsgClient1->ConnectionParams->RemotePort = StrToInt(edPort->Text);
  }
  catch(...)
  {
    ShowMessage("Invalid Port");
    return;
  }
  bnRegister->Enabled = false;
  fmMain->MsgClient1->UserID = MSG_INVALID_USER_ID;
//  fmMain->MsgClient1->UserID = UserInfo->UserID;
  // fill UserInfo
	UserInfo.UserName = (AnsiString)RegUserName->Text;
	UserInfo.FirstName = (AnsiString)RegUserFirstName->Text;
	UserInfo.LastName = (AnsiString)RegUserLastName->Text;
	UserInfo.Organization = (AnsiString)RegUserCompany->Text;
  UserInfo.Department = (AnsiString)RegUserDepartment->Text;
  try
  {
    fmMain->MsgClient1->Connected = true;
  }
  catch (...)
  {
    ShowMessage("Registration failed. Cannot connect to server. Check parameters and ensure that server is started.");
  }
  res = fmMain->MsgClient1->RegisterNewUser(UserInfo,edPassword->Text,true);
  if (res != MSG_COMMAND_OK)
   {
    fmMain->MsgClient1->Connected = false;
    ShowMessage("Registration failed. Error code #"+IntToStr(res)+". Check parameters.");
   }
  else
   {
    fmMain->MsgClient1->Connected = false;
    fmMain->MsgClient1->Active = false;
    fmMain->MsgClient1->UserID = UserInfo.UserID;
    fmMain->MsgClient1->Password = edPassword->Text;
    if (fmMain->Login())
      FClose = true;
    else
      ShowMessage("Login failed. Check parameters and ensure that server is started.");
   }
  bnRegister->Enabled = true;
}
//---------------------------------------------------------------------------

File: MsgCommunicator/Demos/_Databases/Accuracer/Delphi 2009/Client/uRegister.pas
unit uRegister;

interface

{$IFDEF VER200}
{$WARN IMPLICIT_STRING_CAST OFF}
{$WARN IMPLICIT_STRING_CAST_LOSS OFF}
{$ENDIF}

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms,
  Dialogs, StdCtrls, ExtCtrls, MsgComBase, MsgConst;

type
  TfmRegister = class(TForm)
    Panel1: TPanel;
    Panel2: TPanel;
    Label12: TLabel;
    Label10: TLabel;
    Label13: TLabel;
    Label14: TLabel;
    Label15: TLabel;
    Label16: TLabel;
    RegUserDepartment: TEdit;
    RegUserCompany: TEdit;
    RegUserLastName: TEdit;
    RegUserFirstName: TEdit;
    RegUserName: TEdit;
    RegUserID: TEdit;
    bnRegister: TButton;
    bnCancel: TButton;
    edPassword: TEdit;
    Label1: TLabel;
    GroupBox1: TGroupBox;
    Label2: TLabel;
    Label3: TLabel;
    edHost: TEdit;
    edPort: TEdit;
    procedure FormShow(Sender: TObject);
    procedure FormClose(Sender: TObject; var Action: TCloseAction);
    procedure bnCancelClick(Sender: TObject);
    procedure bnRegisterClick(Sender: TObject);
  private
    { Private declarations }
    FClose: Boolean;
  public
    { Public declarations }
  end;

var
  fmRegister: TfmRegister;

implementation

uses uMain, MsgClient;

{$R *.dfm}

procedure TfmRegister.FormShow(Sender: TObject);
begin
  edHost.Text := fmMain.MsgClient1.ConnectionParams.RemoteHost;
  edPort.Text := IntToStr(fmMain.MsgClient1.ConnectionParams.RemotePort);
  FClose := False;
  ModalResult := mrCancel;
end;

procedure TfmRegister.FormClose(Sender: TObject; var Action: TCloseAction);
begin
  if (not FClose) then
   Action := TCloseAction(caNone);
end;

procedure TfmRegister.bnCancelClick(Sender: TObject);
begin
  ModalResult := mrCancel;
  FClose := True;
end;

procedure TfmRegister.bnRegisterClick(Sender: TObject);
var UserInfo: TMsgUserInfo;
    res:      Integer;
begin
  // clear UserInfo
  FillChar(UserInfo,SizeOf(UserInfo),$00);
  ModalResult := mrOK;
  fmMain.MsgClient1.ConnectionParams.RemoteHost := edHost.Text;
  // if UserID empty then UserID will be assigned by the server
  if (RegUserID.Text = '') then
   UserInfo.UserID := MSG_INVALID_USER_ID
  else
   try
    UserInfo.UserID := Cardinal(StrToIntDef(RegUserID.Text,Integer(MSG_INVALID_USER_ID)));
   except
    ShowMessage('Invalid UserID');
    Exit;
   end;
  try
    fmMain.MsgClient1.ConnectionParams.RemotePort := StrToInt(edPort.Text);
  except
    ShowMessage('Invalid Port');
    Exit;
  end;
  bnRegister.Enabled := False;
  fmMain.MsgClient1.UserID := MSG_INVALID_USER_ID;
//  fmMain.MsgClient1.UserID := UserInfo.UserID;
  // fill UserInfo
  UserInfo.UserName := RegUserName.Text;
  UserInfo.FirstName := RegUserFirstName.Text;
  UserInfo.LastName := RegUserLastName.Text;
  UserInfo.Organization := RegUserCompany.Text;
  UserInfo.Department := RegUserDepartment.Text;
  try
    fmMain.MsgClient1.Connected := True;
  except
    ShowMessage('Registration failed. Cannot connect to server. Check parameters and ensure that server is started.')
  end;

  res := fmMain.MsgClient1.RegisterNewUser(UserInfo,edPassword.Text,True);
  if (res <> MSG_COMMAND_OK) then
   begin
    fmMain.MsgClient1.Connected := False;
    ShowMessage('Registration failed. Error code #'+IntToStr(res)+'. Check parameters.');
   end
  else
   begin
    fmMain.MsgClient1.Connected := False;
    fmMain.MsgClient1.Active := False;
    fmMain.MsgClient1.UserID := UserInfo.UserID;
    fmMain.MsgClient1.Password := edPassword.Text;
    if (fmMain.Login) then
      FClose := True
    else
      ShowMessage('Login failed. Check parameters and ensure that server is started.');
   end;
  bnRegister.Enabled := True;
end;

end.
